File: app/src/test/java/com/dominikgaller/alpinebooking/booking/core/domain/BookingIdTest.java
package com.dominikgaller.alpinebooking.booking.core.domain;

import org.junit.jupiter.api.Test;

import java.util.UUID;

import static org.assertj.core.api.Assertions.assertThat;
import static org.assertj.core.api.Assertions.assertThatNullPointerException;

class BookingIdTest {

    @Test
    void constructsWithValidUuid() {
        final UUID uuid = UUID.randomUUID();
        final BookingId id = BookingId.of(uuid);
        assertThat(id.value()).isEqualTo(uuid);
    }

    @Test
    void generateReturnsNonNull() {
        assertThat(BookingId.generate()).isNotNull();
        assertThat(BookingId.generate().value()).isNotNull();
    }

    @Test
    void equalsByValue() {
        final UUID uuid = UUID.randomUUID();
        assertThat(BookingId.of(uuid)).isEqualTo(BookingId.of(uuid));
    }

    @Test
    void nullValueThrows() {
        assertThatNullPointerException().isThrownBy(() -> BookingId.of(null));
    }
}

File: src/test/kotlin/com/example/contractmanagement/booking/core/domain/tourbooking/BookingIdTest.kt
package com.example.contractmanagement.booking.core.domain.tourbooking

import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test
import java.util.UUID

class BookingIdTest {
    @Test
    fun of_wrapsTheGivenValue() {
        val uuid = UUID.fromString("22222222-2222-2222-2222-222222222222")

        assertThat(BookingId.of(uuid).value).isEqualTo(uuid)
    }

    @Test
    fun generate_producesADistinctValue_onEachCall() {
        assertThat(BookingId.generate()).isNotEqualTo(BookingId.generate())
    }

    @Test
    fun equality_isStructural() {
        val uuid = UUID.randomUUID()

        assertThat(BookingId.of(uuid)).isEqualTo(BookingId.of(uuid))
    }
}
